user_service/apis/handlers/user.go

```diff
@@ -38,31 +40,44 @@ func (uh *UserHandlers) HandleGetProfile(ctx *gin.Context) {
 }
 
 func (uh *UserHandlers) HandleLogin(ctx *gin.Context) {
-	var requestBody models.LoginRequestBody
+	var requestBody models.LoginRequest
 
 	if err := ctx.ShouldBindJSON(&requestBody); err != nil {
 		ctx.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
 		return
 	}
-	message := uh.userService.Login(requestBody)
-
+	message, err := uh.userService.Login(requestBody)
+	if err != nil {
+		ctx.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
 	ctx.JSON(http.StatusOK, gin.H{
 		"message": message,
 	})
 }
 
 func (uh *UserHandlers) HandleSignup(ctx *gin.Context) {
-	var requestBody models.SignupRequestBody
+	var requestBody models.SignupRequest
 
 	if err := ctx.ShouldBindJSON(&requestBody); err != nil {
 		ctx.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
 		return
 	}
 
-	message := uh.userService.Signup(requestBody)
+	token, err := uh.userService.Signup(requestBody)
+	if err != nil {
+		ctx.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
+
+	err = uh.emailService.SendEmail(token, requestBody.Email)
+	if err != nil {
+		ctx.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+		return
+	}
 
```

Comment:
I think `SendEmail` is part of SignUp business logic
meaning that, sign up is incomplete w/o sending the email. So the SignUp service should have logic for sending email rather than the handler
wdyt?